hearpreprocess/util/audio.py

```diff
@@ -3,129 +3,168 @@
 """
 
 import json
-import os
-import subprocess
-from collections import Counter
+from collections import Counter, defaultdict
 from pathlib import Path
-from typing import List, Optional, Union
+from typing import List, Optional, Union, Dict
 
 import numpy as np
-import soundfile as sf
+import ffmpeg
 from tqdm import tqdm
 
 
 def mono_wav_and_fix_duration(in_file: str, out_file: str, duration: float):
     """
     Convert to WAV file and trim to be equal to or less than a specific length
+
+    If the audio is already mono and has the required duration, the step is skipped
+    and a symlink is created to the original file
     """
-    ret = subprocess.call(
-        [
-            "ffmpeg",
-            "-y",
-            "-i",
-            str(in_file),
-            "-filter_complex",
-            f"apad=whole_dur={duration},atrim=end={duration}",
-            "-ac",
-            "1",
-            "-c:a",
-            "pcm_f32le",
-            str(out_file),
-        ],
-        stdout=subprocess.DEVNULL,
-        stderr=subprocess.DEVNULL,
-    )
-    # Make sure the return code is 0 and the command was successful.
-    assert ret == 0, f"ret = {ret}"
-
-
-def convert_to_mono_wav(in_file: str, out_file: str):
-    devnull = open(os.devnull, "w")
-    # If we knew the sample rate, we could also pad/trim the audio file now, e.g.:
-    # ffmpeg -i test.webm -filter_complex \
-    #    apad=whole_len=44100,atrim=end_sample=44100 \
-    #    -ac 1 -c:a pcm_f32le ./test.wav
-    # print(" ".join(["ffmpeg", "-y", "-i", in_file,
-    #    "-ac", "1", "-c:a", "pcm_f32le", out_file]))
-    ret = subprocess.call(
-        ["ffmpeg", "-y", "-i", in_file, "-ac", "1", "-c:a", "pcm_f32le", out_file],
-        stdout=devnull,
-        stderr=devnull,
-    )
-    # Make sure the return code is 0 and the command was successful.
-    assert ret == 0
+    # Get the audio stats for the audio file
```

Comment:
Same here and throughout the file

Reply:
This function is no more in use so I removed it. We only use the `mono_wav_and_fix_duration`